### PR TITLE
Speedup quarkus testsuite

### PR DESCRIPTION
**Pull request by DGuhr**
draft PR for getting quarkus tests execution time lower. see commit descriptions for steps.

### CODE REVIEW

**Comment by DGuhr** [comment]
49m. like :)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -436,7 +436,7 @@ jobs:
 
       - name: Run Quarkus Storage Tests
         run: |
-          ./mvnw clean install -nsu -B -f quarkus/tests/pom.xml -Ptest-database -Dtest=PostgreSQLDistTest,MariaDBDistTest#testSuccessful,MySQLDistTest#testSuccessful,DatabaseOptionsDistTest,JPAStoreDistTest,HotRodStoreDistTest,MixedStoreDistTest | misc/log/trimmer.sh
+          ./mvnw clean install -nsu -B -f quarkus/tests/pom.xml -Ptest-database -Dtest=PostgreSQLDistTest#testSuccessful,MySQLDistTest#testSuccessful,DatabaseOptionsDistTest,JPAStoreDistTest,HotRodStoreDistTest,MixedStoreDistTest | misc/log/trimmer.sh
```
Comment (vmuzikar) [inline .github/workflows/ci.yml:439]: Is the removal of MariaDB test here intentional?

**Comment by pedroigor** [comment]
@DGuhr As discussed before, we should now be looking at enabling `chm` when running dist tests.

However, the approach here can be improved to avoid setting the storage on a per-test basis.

**Review by vmuzikar — COMMENTED** [review]
@DGuhr Thank you for the initiative for improving test run times!

In general I like the approach, although I think we should make the storage type generic for whole testsuite, not to specify it at test method basis. At the same time we need to be careful and make sure we don't omit test cases for legacy store.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (DGuhr) [reply .github/workflows/ci.yml:439]: yes, see commit message.

**Comment by DGuhr** [comment]
@pedroigor @vmuzikar  Open for everything. this was just a bit of playing around to see where we land and if it really improves times a good chunk.

**Comment by pedroigor** [comment]
@DGuhr Thanks for the initiative and it would be nice if we could:

* A generic approach to enable `chm` by default when running the test suite and review tests accordingly when legacy is required
* A generic approach to enable `cache=local` by default when running the start command and review tests accordingly when clustering is required
* Review tests relying on the test execution order and inject the distribution instead so that we can perform multiple sequential steps from a single method
* Make sure we don't run JVM tests in CI, mainly those extended to run using the distribution
* Right now, postgres is our target database (considering the new store) and we should have *all* tests enabled for it, not only check for success. Let us also keep MariaDB and include MySQL as proposed. More test coverage is better and since we are discussing improving execution time, it should help improve coverage via GHA.
* As you pointed out we can and we should review tests that might be "re-installing" the raw distribution unnecessarily

Please, let me know if you need any help.

**Comment by pedroigor** [comment]
Thanks. Closing as we need more discussions around the points herein mentioned.

**Pull request closed without merging**